src/server/middleware/blob/BlobServer.js

```diff
@@ -24,9 +25,19 @@ function createExpressBlob(options) {
     ASSERT(typeof options.gmeConfig !== 'undefined', 'gmeConfig required');
     ASSERT(typeof options.ensureAuthenticated === 'function', 'ensureAuthenticated must be given.');
 
-    ensureAuthenticated = options.ensureAuthenticated;
+    accessTokens = options.accessTokens;
     getUserId = options.getUserId;
     logger = options.logger.fork('middleware:BlobServer');
+    ensureAuthenticated = function (req, res, next) {
```

Comment:
maybe I miss the point, but how is this different from simply calling the regular ensure authenticated? (no reference for the accessToken or not getting the userId from it...)

Reply:
`ensureAuthenticated` assumes the user is not authenticated. That is, if we authenticate with a token, it will return a `403` since no other authentication techniques pass. This function only calls `ensureAuthenticated` if the user does not have a valid user ID after authenticating with the access token.

Reply:
I have been a little torn about how deeply to integrate the token authentication since it would not surprise me if people wanted fine-grained permissions from them later. This would make authentication from within `ensureAuthenticated` more challenging as the authentication would depend on the current scope of the access token and the function does not currently have any concept of the scope or feature that is is authenticating for.

However, it could still be useful to update `ensureAuthenticated` to early return if the user is already authenticated...

Reply:
So I agree that it should not be too tightly integrated as we are not sure of the complete extent on how the access token can be fine tuned in the future.
Still, in the current form, you just check it the userId is placed in the request, which in my book not equal to checking it the request has a valid access token. I mean I agree that the regular checking should only be called if the accessToken is not there or not verified...
(again it could be that I missed somehting, but it would be more clean if the ensure authenticated would show that we verify the accesstoken and then we go on for the original method...